cmake/BuildLibCURL.cmake

```diff
@@ -142,6 +146,37 @@ macro(curl_build)
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_POP3=ON")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_IMAP=ON")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_SMTP=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_ALTSVC=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_DOH=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_GETOPTIONS=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_HSTS=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_HTTP_AUTH=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_LIBCURL_OPTION=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_MIME=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_MQTT=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_NETRC=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_NTLM=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_OPENSSL_AUTO_LOAD_CONFIG=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_PARSEDATE=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_PROGRESS_METER=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_PROXY=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_SHUFFLE_DNS=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_SOCKETPAIR=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_DISABLE_VERBOSE_STRINGS=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_ENABLE_EXPORT_TARGET=ON")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_ENABLE_SSL=ON")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_LTO=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_BEARSSL=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_GSSAPI=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_LIBSSH=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_OPENLDAP=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_WERROR=OFF")
+    list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_ZSTD=OFF")
```

Comment:
I think no. Zlib is already enabled for curl (see option `-DCURL_ZLIB=ON` above). I'm not sure that replacing Zlib to Zstd makes sense.

Reply:
Isn't it about HTTP content encoding (gzip, brotli and so on)? If so, it worth to enable. When we initially bundle it, I enabled gzip (because it does not add new deps) and disabled brotli (because it would require one more dependency). But this time curl had no zstd content encoding support.

Reply:
BTW, zstd symbols in tarantool are prefixed with `tnt_`, so extra work may be necessary to let curl use them.

Reply:
Yep, it is really about content encoding. It would be nice to have, but since it requires more work, I would postpone it.

I propose to add related information to a comment in the code and go ahead.

Reply:
Yes, it is about content encoding.

```
commit e13357b14b7de013043d704dc9c80697e62616c6
Author: Gilles Vollant <info@winimage.com>
Date:   Mon May 25 17:49:46 2020 +0200

    content_encoding: add zstd decoding support
    
    include zstd curl patch for Makefile.m32 from vszakats
    and include Add CMake support for zstd from Peter Wu
    
    Helped-by: Viktor Szakats
    Helped-by: Peter Wu
    Closes #5453
```

Reply:
Added a note:
```diff
--- a/cmake/BuildLibCURL.cmake
+++ b/cmake/BuildLibCURL.cmake
@@ -176,6 +176,10 @@ macro(curl_build)
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_LIBSSH=OFF")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_OPENLDAP=OFF")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_WERROR=OFF")
+    # CURL_ZSTD adds zstd encoding/decoding support. Switching on probably will
+    # require zstd symbols in Tarantool are prefixed with 'tnt_', so switching
+    # CURL_ZSTD on may require extra work to let Curl use them, and we should
+    # also pass correct include directory.
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_ZSTD=OFF")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DLIBCURL_OUTPUT_NAME=libcurl")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DENABLE_CURLDEBUG=OFF")

```

Reply:
> will require zstd symbols in Tarantool are prefixed with 'tnt_'

Symbols in tarantool are already prefixed with `tnt_`. Tuning libcurl's build to catch the symbols may require extra work.

(It is minor comment, not a blocker.)

Reply:
```diff
+++ b/cmake/BuildLibCURL.cmake
@@ -176,10 +176,8 @@ macro(curl_build)
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_LIBSSH=OFF")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_USE_OPENLDAP=OFF")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_WERROR=OFF")
-    # CURL_ZSTD adds zstd encoding/decoding support. Switching on probably will
-    # require zstd symbols in Tarantool are prefixed with 'tnt_', so switching
-    # CURL_ZSTD on may require extra work to let Curl use them, and we should
-    # also pass correct include directory.
+    # CURL_ZSTD adds zstd encoding/decoding support. Tuning libcurl's build to
+    # catch the symbols may require extra work.
     list(APPEND LIBCURL_CMAKE_FLAGS "-DCURL_ZSTD=OFF")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DLIBCURL_OUTPUT_NAME=libcurl")
     list(APPEND LIBCURL_CMAKE_FLAGS "-DENABLE_CURLDEBUG=OFF")
```